### PR TITLE
Add Ruby 3.1 to the CI matrix

### PR DESCRIPTION
**Pull request by petergoldstein**
This PR just adds Ruby 3.1 to the CI matrix.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: ["3.0", "2.7", "2.6", "2.5"]
+        ruby: ["3.1", "3.0", "2.7", "2.6", "2.5"]
```
Comment (OsamaSayegh) [inline .github/workflows/ci.yml:25]:
```suggestion
        ruby: ["3.1", "3.0", "2.7", "2.6"]
```

2.5 is EOL which we no longer support.

Comment (petergoldstein) [reply .github/workflows/ci.yml]: Sure.  Removed.

**Comment by petergoldstein** [comment]
@OsamaSayegh I've removed 2.5

**Comment by OsamaSayegh** [comment]
Awesome, thank you!

**Pull request merged**